### ISSUE TITLE
Not all the data from DataFrame got write to Redis

### ISSUE THREAD

**Issue by zhysn1201** [RedisLabs/spark-redis#347]
I have built a DataFrame roughly has 2K records, and each record has a timezone field and a id field where id is about 240,000 string length. 

When I call 
`df.write
      .format("org.apache.spark.sql.redis")
      .option("table", s"push:candidates:${date}")
      .option("key.column", "time_zone_indexed")
      .mode(SaveMode.Append)
      .save()`

I observed that not all the data have been saved into Redis. Nearly 500 keys have been written in the Redis. Do you know why?

Things I have tried:
1. Extend Redis timeout to 10 mins
2. Extend `spark.redis.max.pipeline.size` to 200

**Comment by fe2s** [comment]
Did it fail? Any exceptions?

**Comment by zhysn1201** [comment]
Nope. The spark job seems succeeded. I do not see any exceptions from the internal Spark portal. However, I am not sure if there is any other places that I can check the logs. Spark UI?

**Comment by fe2s** [comment]
I would try to check the driver and executor logs if there are any exceptions. You may also want to check the number of unique keys in your dataframe.

**Comment by dharmesh0500** [comment]
Hello @fe2s 
I am also facing the same issue while using toRedisByteLISTs
my RDD has 26M records but after writing there is always around 16-22M records
is there any config which I can use for consistency?
